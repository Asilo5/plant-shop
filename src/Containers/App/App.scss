main {

  header {
    background-image: url('https://images.unsplash.com/photo-1477554193778-9562c28588c0?ixlib=rb-1.2.1&ixid=eyJhcHBfaWQiOjEyMDd9&auto=format&fit=crop&w=800&q=60');
    background-size: cover;
    background-repeat: no-repeat;
    height: 80vh;
    h2 {
      display: flex;
      justify-content: center;
      align-content: center;
      font-size: 5em;
      margin: 0;
      color: white;
    }
  }
}
