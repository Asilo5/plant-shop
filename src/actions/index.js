export const addPlants = (setPlants) => ({
    type: 'ADD_PLANTS',
    setPlants
});

export const hasError = (error) => ({
  type: 'HAS_ERROR',
  error
});

export const newPlant = (plant) => ({
  type: 'NEW_PLANT',
  plant
})