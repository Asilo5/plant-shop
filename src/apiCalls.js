export const getPlants = async () => {

      const resp = await fetch('http://localhost:3001');
      const data = await resp.json();

      if(!resp.ok) {
        throw new Error('No Plants Found!');
      } else {
        return data;
      }
   
}