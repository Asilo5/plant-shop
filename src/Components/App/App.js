import React, { Component } from 'react';
import './App.scss';
import  { getPlants } from '../../apiCalls';

class App extends Component  {
   
  componentDidMount = () => {
    getPlants()
      .then(plants => console.log(plants))
      .catch(err => console.log(err))
  }

  render() {
    return (
      <section>
         <p>Plants</p>
      </section>
    )
  }
}

export default App;
