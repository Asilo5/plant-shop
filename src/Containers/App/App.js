import React, { Component } from 'react';
import './App.scss';
import  { getPlants, deletePlantFetch } from '../../apiCalls';
import { addPlants, hasError, deletePlant } from '../../actions';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import { Route } from 'react-router-dom';
import PropTypes from 'prop-types';

import NavBar from '../../Components/NavBar/NavBar';
import ViewPlant from '../ViewPlant/ViewPlant';
import PlantsContainer from '../PlantsContainer/PlantsContainer';
import AddPlants from '../AddPlants/AddPlants';
import FavouritesContainer from '../FavouritesContainer/FavouritesContainer';
import Header from '../../Components/Header/Header';

export class App extends Component  {
   
  componentDidMount = () => {
    const { addPlants, hasError } = this.props;
    getPlants()
      .then(plants => addPlants(plants.plants))
      .catch(err => hasError(err.error))
  }

  deleteThePlant = (id) => {
    const { deletePlant, hasError } = this.props;
    deletePlantFetch(id)
      .then(plants => {
        console.log(plants)
        deletePlant(plants.plants._id)
      })
      .catch(err => hasError(err.error))
  }

  render() {
    return (
      <main>
         <NavBar />
         <Route exact path='/' component={Header} />
         <Route exact path='/allPlants' render={() => <PlantsContainer deleteThePlant={this.deleteThePlant} />} />
         <Route exact path='/plant/:id' render={ ({ match }) => {
           const { id } = match.params;
           const { allPlants } = this.props;

           let plantsList = allPlants.find((plant) => plant._id === id);

           return <ViewPlant {...plantsList} deleteThePlant={this.deleteThePlant} />
         } } />
         <Route exact path='/addPlants' component={AddPlants} />
         <Route exact path='/lovedPlants' component={FavouritesContainer} />
      </main>
    )
  }
}

export const mapStateToProps = (state) => ({
  allPlants: state.allPlants
})

export const mapDispatchToProps = (dispatch) => (
  bindActionCreators({
    addPlants: setPlants => dispatch( addPlants(setPlants) ),
    hasError: error => dispatch( hasError(error) ),
    deletePlant: id => dispatch( deletePlant(id) )
  }, dispatch)
)

App.propTypes = {
  allPlants: PropTypes.array,
  addPlants: PropTypes.func, 
  hasError: PropTypes.func,
  deletePlant: PropTypes.func
}

export default connect(mapStateToProps, mapDispatchToProps)(App);
