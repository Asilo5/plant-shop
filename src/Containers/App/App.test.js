import React from 'react';
import { App, mapDispatchToProps } from './App';
import { shallow } from 'enzyme';
import { addPlants } from '../../actions';
import  { getPlants } from '../../apiCalls';

jest.mock('../../apiCalls');

describe('App container', () => {

  describe('App', () => {

    let wrapper;
    let mockPlants = [
      {
        "_id": "5e16b29d88553086edcddc53",
        "name": "Aloe Queen",
        "description": "Healing plant",
        "care": "Lots of Love",
        "price": 20,
        "image": "https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg",
        "__v": 0
      }
    ];

     beforeEach(() => {
      getPlants.mockImplementation(() => {
        return Promise.resolve(mockPlants)
      });

       wrapper = shallow(<App addPlants={jest.fn()} /> )
     })

     it('should match snapshot with all information passing in correctly', () => {
       expect(wrapper).toMatchSnapshot();
     })

     it('should call getMovies after component is mounted', () => {
      expect(getPlants).toHaveBeenCalled();
    })

  })

  describe('mapDispatchToProps', () => {
    let mockPlants = [
      {
        "_id": "5e16b29d88553086edcddc53",
        "name": "Aloe Queen",
        "description": "Healing plant",
        "care": "Lots of Love",
        "price": 20,
        "image": "https://target.scene7.com/is/image/Target/GUEST_be7e6313-15ac-4e1c-94b8-8af5595d6cb6?wid=488&hei=488&fmt=pjpeg",
        "__v": 0
      }
    ];

    const mockDispatch = jest.fn();

    it('should dispatch addPlants', () => {
      const actionToDispatch = addPlants(mockPlants);
      const mappedProps = mapDispatchToProps(mockDispatch);

      mappedProps.addPlants(mockPlants);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    })
  })
   
})


