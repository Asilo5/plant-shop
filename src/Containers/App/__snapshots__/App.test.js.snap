// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App container App should match snapshot with all information passing in correctly 1`] = `
<main>
  <NavBar />
  <Route
    component={[Function]}
    exact={true}
    path="/"
  />
  <Route
    exact={true}
    path="/allPlants"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/plant/:id"
    render={[Function]}
  />
  <Route
    component={
      Object {
        "$$typeof": Symbol(react.memo),
        "WrappedComponent": [Function],
        "compare": null,
        "displayName": "Connect(AddPlants)",
        "type": [Function],
      }
    }
    exact={true}
    path="/addPlants"
  />
</main>
`;
